src/Sparrow.Server/Utils/DiskStatsGetter/LinuxDiskStatsGetter.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Text;
+using Mono.Unix.Native;
+using Sparrow.Logging;
+
+namespace Sparrow.Server.Utils.DiskStatsGetter;
+
+[SupportedOSPlatform("linux")]
+internal class LinuxDiskStatsGetter : DiskStatsGetter<LinuxDiskStatsRawResult>
+{
+    private static readonly Logger Logger = LoggingSource.Instance.GetLogger<LinuxDiskStatsGetter>("Server");
+
+    public LinuxDiskStatsGetter(TimeSpan minInterval) : base(minInterval)
+    {
+    }
+
+    protected override DiskStatsResult CalculateStats(LinuxDiskStatsRawResult currentInfo, State state)
+    {
+        var diff = (currentInfo.Time - state.Prev.Raw.Time).TotalSeconds;
+        var diskSpaceResult = new DiskStatsResult
+        {
+            IoReadOperations = (currentInfo.IoReadOperations - state.Prev.Raw.IoReadOperations) / diff, 
+            IoWriteOperations = (currentInfo.IoWriteOperations - state.Prev.Raw.IoWriteOperations) / diff, 
+            ReadThroughput = new Size((long)((currentInfo.ReadSectors - state.Prev.Raw.ReadSectors) * 512 / diff), SizeUnit.Bytes), 
+            WriteThroughput = new Size((long)((currentInfo.WriteSectors - state.Prev.Raw.WriteSectors) * 512 / diff), SizeUnit.Bytes), 
+            QueueLength = currentInfo.QueueLength
+        };
+        return diskSpaceResult;
+    }
+
+    protected override LinuxDiskStatsRawResult GetDiskInfo(string path)
+    {
+        try
+        {
+            var (major, minor) = GetDiskMajorMinor(path);
+
+            var statPath = $"/sys/dev/block/{major}:{minor}/stat";
+            using var reader = File.OpenRead(statPath);
+            
+            return ReadParse(reader);
+        }
+        catch (Exception e)
+        {
+            if(Logger.IsInfoEnabled)
+                Logger.Info($"Could not get GetDiskInfo of {path}", e);
+            return null;
+        }
+    }
+        
+    private static (ulong Major, ulong Minor) GetDiskMajorMinor(string path)
+    {
+        if (Syscall.stat(path, out var stats) != 0)
+        {
+            var errno = Stdlib.GetLastError();
+            var errorBuilder = new StringBuilder();
+            Syscall.strerror_r(errno, errorBuilder,1024);
```

Comment:
*Pretty* sure that this will lead to a heap corruption. You are passing a `StringBuilder`, but it didn't actually allocate any buffer here, did it?
There is a default capacity of 16 chars, I think. And if you have more, it will just overwrite those.

Reply:
Aware that this is in the original code as well

Reply:
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1838

Let's allocate that on the stack, instead. And use it properly

Reply:
This is a third-party API - `Mono.Unix.Native.Syscall`.
Do you want me to stop using it?
Or just iniail the StringBuilder with capacity

Reply:
You need to initialize with capacity, yes

Reply:
Meanwhile, I set the capacity